### ISSUE TITLE
Failed tab does not count "Too Many Errors, Failing" as Failed WU

### ISSUE THREAD

**Issue by muziqaz** [harlam357/hfm-net#348]
HI,

I noticed that this sequence:

> 19:01:34:WARNING:WU01:FS00:FahCore returned an unknown error code which probably indicates that it crashed
> 19:01:34:WARNING:WU01:FS00:FahCore returned: UNKNOWN_ENUM (-1073740791 = 0xc0000409)
> 19:01:38:WARNING:WU01:FS00:FahCore returned an unknown error code which probably indicates that it crashed
> 19:01:38:WARNING:WU01:FS00:FahCore returned: UNKNOWN_ENUM (-1073740791 = 0xc0000409)
> 19:02:37:WARNING:WU01:FS00:FahCore returned an unknown error code which probably indicates that it crashed
> 19:02:37:WARNING:WU01:FS00:FahCore returned: UNKNOWN_ENUM (-1073740791 = 0xc0000409)
> 19:02:37:WARNING:WU01:FS00:Too many errors, failing
> 19:01:34:WARNING:WU01:FS00:FahCore returned an unknown error code which probably indicates that it crashed
> 19:01:34:WARNING:WU01:FS00:FahCore returned: UNKNOWN_ENUM (-1073740791 = 0xc0000409)
> 19:01:38:WARNING:WU01:FS00:FahCore returned an unknown error code which probably indicates that it crashed
> 19:01:38:WARNING:WU01:FS00:FahCore returned: UNKNOWN_ENUM (-1073740791 = 0xc0000409)
> 19:02:37:WARNING:WU01:FS00:FahCore returned an unknown error code which probably indicates that it crashed
> 19:02:37:WARNING:WU01:FS00:FahCore returned: UNKNOWN_ENUM (-1073740791 = 0xc0000409)
> 19:02:37:WARNING:WU01:FS00:Too many errors, failing

does not add a failed count to Fail tab in main page.
The sequence above is extract from failing WU. It fails with Unknown_ENUM error, restarts, fails again, restarts and if it keeps failing 5 times (or whatever project owner sets it to), it is marked as failed and sent back as bad WU. HFM does not recognize this as failed WU, however WU History section records it as failed WU.
In order for us to easily catch these failures during the day, when we are at work, it would be great if HFM could recognize this sequence as a failure and add Failed count.
So from this compacted sequence that slot should have 2 failed WUs in main HFM window Failed tab. Instead it still shows 0.
Thanks
Regards

**Comment by harlam357** [comment]
Please send me the complete log for this client containing this sequence of events.

**Comment by harlam357** [comment]
Does WU history show the WU result as UNKNOWN_ENUM?  The completed and failed counts are sourced from WU history and the failed count query checks for WUs where the result is not FINISHED_UNIT.

**Comment by muziqaz** [comment]
[log-20210525-225905.txt](https://github.com/harlam357/hfm-net/files/6544377/log-20210525-225905.txt)

Here is a log. I will check out the history once I'm at home.

Regards

**Comment by muziqaz** [comment]
Ok, update. It seems that HFM does not even record these failed WUs in the history at all :/

**Comment by harlam357** [comment]
Makes sense since unknown enum is not a definitive result of the WU.  Will have to key on "Too many errors, failing".
